### PR TITLE
emptyPageDiagnsticsShouldOnlyControlLogging

### PR DESCRIPTION
**Pull request by xinlian12**
`emptyPageDiagnosticsEnabled` should only control the empty page logging. Customer can toggle the system property to configure the intentions, it should not be override based on whether metrics being enabled.

### CODE REVIEW

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #### Breaking Changes
 
 #### Bugs Fixed
+* Fixed an issue where `emptyPageDiagnosticsEnabled` is being overriden with different meaning - this causes empty page diagnostics being logged (with INFO level) when they shouldn't - See [PR 37199](https://github.com/Azure/azure-sdk-for-java/pull/37199)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:10]:
```suggestion
* Fixed an issue where `emptyPageDiagnosticsEnabled` in `CosmosQueryRequestOptions` was being overridden. This caused empty page diagnostics to be logged (with INFO level) even when the flag was set to false - See [PR 37199](https://github.com/Azure/azure-sdk-for-java/pull/37199)
```

**Review by kushagraThapar — APPROVED** [review]
LGTM, however, would be good to have a test case for this if possible.

**Comment by xinlian12** [comment]
All the tests have passed before making the changelog update.

**Pull request merged**